main.cpp

```diff
@@ -109,18 +119,34 @@ int main()
     /*
      construct 2 more Car instances via emplace_back.
      */
-    
+    cars.emplace_back("Ford");
+    cars.emplace_back("Mercedes");
     /*
      now reserve and emplace_back your Trucks and Motorcyles
      */
-    
-    
-    
-    
-    assert(false);
+
+     motorcycles.emplace_back("Ducati");
+     motorcycles.emplace_back("Harley Davidson");
+
+     trucks.emplace_back("SemiTruck");
+
+    for(Car& car : cars)
+    {
+        highway.addVehicle(&car);
+    }
+
+    for(Motorcycle& mc : motorcycles)
+    {
+        highway.addVehicle(&mc);
+    }
+
+    for(SemiTruck& st : trucks)
+    {
+        highway.addVehicle(&st);
+    }
```

Comment:
use `auto&` instead of specific types in case your container changes what it owns.
- [ ] revise